harness/determined/pytorch/_trainer.py

```diff
@@ -159,8 +151,8 @@ def fit(
             if not isinstance(max_length.value, int):
                 raise TypeError("max_length must be configured in TrainUnit(int) types.")
 
-            if self._det_profiler:
-                logger.warning("Determined profiler will be ignored in local training mode.")
+            if profiling_enabled:
```

Comment:
For me and my own understanding trying to work this out, if `self._local_training`, then the `core_context` in the `PyTorchTrialController` this creates will have a `DistributedContext` without any rank information, and so the `core_context.profiler` won't (effectively) be turned on?

Reply:
no. local training mode means the core context will get initialized with `dummy_init` and a `DummyProfilerContext` will get created, which is effectively a no-op.

Reply:
Via the `task_type == "TRIAL"` test here?

https://github.com/determined-ai/determined/blob/c5cf597d35188aa05e17fb03ddc66b062864e0a2/harness/determined/core/_context.py#L255

Reply:
Aha! Yes!

https://github.com/determined-ai/determined/blob/c5cf597d35188aa05e17fb03ddc66b062864e0a2/harness/determined/pytorch/_trainer.py#L33

Reply:
I feel like there's been a suboptimal decision here in the past. One that has nothing to do with this PR whatsoever.

This location and the core context are both making decisions based on `task_type == TRIAL`, and each have to guess at what the other's decided. If you feel like talking about it sometime, could you chat at me what a better design would have been? My hope is that a conversation like that would teach me about the harness.

Reply:
>Via the task_type == "TRIAL" test here?

no, the dummy context gets created before that, at:
```
    info = det.get_cluster_info()
    if info is None:
        return _dummy_init(
            distributed=distributed,
            checkpoint_storage=checkpoint_storage,
        )
```

> each have to guess at what the other's decided

i'm not sure what you mean by this exactly. they're both calling `det.get_cluster_info()` but that's separate from both the core context and Trainer API. they both make an independent decision about whether they are running on cluster or not and what to do with that information. core API only cares about whether it's running on a cluster or not, trainer API has its own concept of what local training means.

i do think the whole `local_training` codepath is a bit wonky in Trainer API, maybe that's the feeling you're getting at. the ideal world is for that `if local_training/else` conditional to not exist -- the same code should be able to be run locally and submitted to the cluster. that was one of my main goals with Trainer API. however, our system makes some inflexible assumptions (mostly concerning the searcher) that are baked-in and requires Trainer API to differentiate between local and non-local mode.

in any case, happy to chat about it later, but yeah, definitely out of scope for this PR.